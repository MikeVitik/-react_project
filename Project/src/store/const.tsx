export const BREAK_TIME = 10 * 1000;
export const TASK_TIME = 20 * 1000;
export const TIME_OF_ONE_POMODORO = 25 * 60 * 1000;

export const TOTAL_MILLISECONDS_IN_DAY = 1000 * 60 * 60 * 24;
