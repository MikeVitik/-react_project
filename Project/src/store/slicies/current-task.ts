import { PayloadAction, createSlice } from "@reduxjs/toolkit";

export interface CurrentTask {
  taskId?: number;
  state:
    | "inited" // нет задач
    | "workInited" // задача есть но она не начата
    | "workTimer" // работаем над задачей
    | "workPause" // работа на паузе
    | "workStop" // после паузы выбор продолжаем или сделали
    | "break" // пауза не начата
    | "breakTimer" // пауза
    | "breakStop"; // пауза остановлена
  hasPause?: boolean;
  startTime?: Date;
  pause?: number;
}
export const currentTaskSlice = createSlice({
  name: "currentTask",
  initialState: {
    state: "inited",
    hasPause: false,
  } satisfies CurrentTask as CurrentTask,
  reducers: {
    initTask: (state, action: { payload: number | undefined }) => {
      if (state.state === "inited") {
        state.state = "workInited";
        state.taskId = action.payload;
      }
    },
    startTask: (state) => {
      if (state.state === "workInited") {
        state.state = "workTimer";
      } else {
        throw new Error("Wrong state:" + state.state);
      }
    },
    pauseTask: (state) => {
      state.state = "workPause";
      state.hasPause = true;
    },
    stopTask: (state) => {
      state.state = "inited";
    },
    continueTask: (state) => {
      state.state = "workTimer";
    },
    stopTaskWork: (state) => {
      state.state = "workStop";
    },
    startBreak: (state) => {
      state.state = "break";
    },
    pauseBreak: (state) => {
      state.state = "breakStop";
    },
    continueBreak: (state) => {
      state.state = "breakTimer";
    },
    nextTaskOrInited: (state) => {
      if (state.taskId !== undefined) {
        state.state = "workInited";
      } else {
        state.state = "inited";
      }
    },
    completeTask: (state, action: PayloadAction<number | undefined>) => {
      state.taskId = action.payload;
      state.state = "break";
    },
  },
  selectors: {
    getMode: (state) => {
      const stateToModeMap: {
        [key in CurrentTask["state"]]: "prepare" | "work" | "pause";
      } = {
        inited: "prepare",
        break: "pause",
        breakStop: "pause",
        breakTimer: "pause",
        workInited: "prepare",
        workPause: "work",
        workStop: "work",
        workTimer: "work",
      };
      return stateToModeMap[state.state];
    },
  },
});

export const currentTask = currentTaskSlice.selectSlice;
