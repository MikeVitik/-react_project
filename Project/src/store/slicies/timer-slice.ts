import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { getTimeFractions } from "../utils/format-time";

export interface Timer {
  state: "created" | "running" | "pause" | "end";
  time: number;
  totalTime: number;
  currentTime: number;
}

export const timerSlice = createSlice({
  name: "timer",
  initialState: {
    state: "end",
    totalTime: 0,
    time: 0,
    currentTime: 0,
  } satisfies Timer as Timer,
  reducers: {
    create: (state, action: PayloadAction<number>) => {
      return {
        isRunning: false,
        time: 0,
        totalTime: action.payload,
        currentTime: 0,
        state: "created",
      };
    },
    start: (state) => {
      state.state = "running";
      state.currentTime = 0;
    },
    stop: (state) => {
      state.state = "pause";
      state.currentTime = 0;
    },
    updateTime: {
      prepare: (delta: number) => ({ payload: { delta } }),
      reducer: (state, action: PayloadAction<{ delta: number }>) => {
        if (state.state === "pause" || state.state === "running") {
          state.currentTime = state.currentTime + action.payload.delta;
        }
        if (state.state === "running") {
          const newTime = state.time + action.payload.delta;
          if (newTime >= state.totalTime) {
            state.state = "end";
          }
          state.time = newTime;
        } else {
          return state;
        }
      },
    },
  },
  selectors: {
    getTimerValue: ({ totalTime, time }) => {
      const elapsedTime = new Date(totalTime - time);
      const { m, s } = getTimeFractions(elapsedTime);
      return { m, s: s < 10 ? "0" + s : s };
    },
  },
});

export const { getTimerValue } = timerSlice.selectors;
