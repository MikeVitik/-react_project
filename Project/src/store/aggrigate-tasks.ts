import { Task } from "./slicies/tasks-slice";
import { normalizeDay } from "./utils/normalize-day";
export type FilterType = "currentWeek" | "pastWeek" | "twoWeeksAgo";
export interface StatisticValues {
  day: number;
  completedPomodoro: number;
  workTime: number;
  pauseCount: number;
  pauseTime: number;
}

export const aggrigateTasks = (tasks: Task[]): StatisticValues[] => {
  const result = Object.values(
    tasks
      .sort((a, b) => a.startDate.getTime() - b.startDate.getTime())
      .reduce<{
        [key: string]: StatisticValues;
      }>((aggrigate, task) => {
        const key = normalizeDay(task.startDate.getDay());
        if (aggrigate[key]) {
          aggrigate[key] = {
            day: key,
            completedPomodoro:
              aggrigate[key].completedPomodoro + task.completedPomodoro! / 1000,
            workTime: aggrigate[key].workTime + task.workTime! / 1000,
            pauseCount: aggrigate[key].pauseCount + task.pauseCount!,
            pauseTime: Math.floor(
              aggrigate[key].pauseTime + task.pauseTime! / 1000
            ),
          };
        } else {
          aggrigate[key] = {
            day: key,
            completedPomodoro: task.completedPomodoro! / 1000,
            workTime: task.workTime! / 1000,
            pauseCount: task.pauseCount!,
            pauseTime: Math.floor(task.pauseTime! / 1000),
          };
        }
        return aggrigate;
      }, {})
  );
  const dayRemains = 7 - result.length;
  if (dayRemains > 0) {
    const days = [...Array(dayRemains)].map((_, i) => ({
      day: i + 1,
      completedPomodoro: 0,
      workTime: 0,
      pauseCount: 0,
      pauseTime: 0,
    }));
    return [...result, ...days];
  }
  return result;
};
