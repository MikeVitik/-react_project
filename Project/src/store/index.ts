import { taskEditorSlice } from "./slicies/task-editor";

import { tasksSlice } from "./slicies/tasks-slice";

import { currentTaskSlice } from "./slicies/current-task";
import { timerSlice } from "./slicies/timer-slice";

export type { CurrentTask } from "./slicies/current-task";

export const { getMode } = currentTaskSlice.selectors;

export { aggrigateStatistic, filterStatistic } from "./slicies/statistic";
export type { FilterType, StatisticValues } from "./slicies/statistic";

export { configureAppStore, useAppDispatch } from "./store";

export { currentTask } from "./selectors/current-task-selector";
export { taskActions } from "./selectors/task-actions-selector";

export type { RootState } from "./store";
export const {
  decrementPomodoro,
  deleteTask,
  incrementPomodoro,
  changeTask,
  addTask,
} = tasksSlice.actions;

export { createTask } from "./actions/task-actions";
export { updateTime } from "./actions/update-time";
export const { selectCanDecrement, selectTotalTime } = tasksSlice.selectors;

export const { cancelTaskEdit, editTask, updateEditTaskName } =
  taskEditorSlice.actions;
export const { getTimerValue } = timerSlice.selectors;

export { generateStatistics } from "./utils/statistic-generator";
