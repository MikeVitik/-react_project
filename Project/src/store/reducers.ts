import { combineSlices } from "@reduxjs/toolkit";
import { currentTaskSlice } from "./slicies/current-task";
import { taskEditorSlice } from "./slicies/task-editor";
import { tasksSlice } from "./slicies/tasks-slice";
import { timerSlice } from "./slicies/timer-slice";

export const rootReducer = combineSlices(
  taskEditorSlice,
  tasksSlice,
  currentTaskSlice,
  timerSlice
);
