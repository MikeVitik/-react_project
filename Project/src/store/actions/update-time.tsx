import { createAsyncThunk } from "@reduxjs/toolkit";
import { currentTaskSlice } from "../slicies/current-task";
import { timerSlice } from "../slicies/timer-slice";
import { RootState } from "../store";
import { pauseBreak, stopTaskWork } from "./task-actions";

export const updateTime = createAsyncThunk<void, number, { state: RootState }>(
  "updateTimer",
  (delta: number, { dispatch, getState }) => {
    dispatch(timerSlice.actions.updateTime(delta));
    if (timerSlice.selectSlice(getState()).state === "end") {
      if (currentTaskSlice.selectSlice(getState()).state === "breakTimer") {
        dispatch(pauseBreak());
      } else {
        dispatch(stopTaskWork());
      }
    }
  }
);
