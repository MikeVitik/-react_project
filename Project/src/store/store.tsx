import {
  ActionCreator,
  configureStore,
  createAsyncThunk,
  createListenerMiddleware,
  createSelector,
  isAnyOf,
} from "@reduxjs/toolkit";
import { useDispatch } from "react-redux";
import { rootReducer } from "./reducers";
import { CurrentTask, currentTaskSlice } from "./slicies/current-task";
import { SerialiazableTask, addTask, tasksSlice } from "./slicies/tasks-slice";
import { timerSlice } from "./slicies/timer-slice";
export const updateTime = createAsyncThunk<void, number, { state: RootState }>(
  "updateTimer",
  (delta: number, thunkAPI) => {
    thunkAPI.dispatch(timerSlice.actions.updateTime(delta));
    if (timerSlice.selectSlice(thunkAPI.getState()).state === "end") {
      // thunkAPI.dispatch(timerEnd)
    }
  }
);

const BREAK_TIME = 10 * 1000;
const TASK_TIME = 20 * 1000;

export const createTask = createAsyncThunk<void, string, { state: RootState }>(
  "createTask",
  (text, { dispatch, getState }) => {
    const addTaskAction = addTask(text);
    dispatch(addTaskAction);
    if (currentTask(getState()).taskId === undefined) {
      dispatch(currentTaskSlice.actions.initTask(addTaskAction.payload.id));
      dispatch(timerSlice.actions.create(TASK_TIME));
    }
  }
);

export const startTask = createAsyncThunk<void, void, { state: RootState }>(
  "startTask",
  (_, { dispatch }) => {
    dispatch(currentTaskSlice.actions.startTask());
    dispatch(timerSlice.actions.start());
  }
);

export const pauseTask = createAsyncThunk<void, void, { state: RootState }>(
  "pauseTask",
  (_, { dispatch }) => {
    dispatch(currentTaskSlice.actions.pauseTask());
    dispatch(timerSlice.actions.stop());
  }
);
export const stopTask = createAsyncThunk<void, void, { state: RootState }>(
  "stopTask",
  (_, { dispatch }) => {
    dispatch(currentTaskSlice.actions.stopTask());
    dispatch(timerSlice.actions.stop());
  }
);
export const continueTask = createAsyncThunk<void, void, { state: RootState }>(
  "continueTask",
  (_, { dispatch }) => {
    dispatch(currentTaskSlice.actions.continueTask());
    dispatch(timerSlice.actions.start());
  }
);

export const startBreak = createAsyncThunk<void, void, { state: RootState }>(
  "startBreak",
  (_, { dispatch, getState }) => {
    dispatch(currentTaskSlice.actions.startBreak());
    dispatch(timerSlice.actions.create(BREAK_TIME));
  }
);

export const pauseBreak = createAsyncThunk<void, void, { state: RootState }>(
  "pauseBreak",
  (_, { dispatch, getState }) => {
    dispatch(currentTaskSlice.actions.pauseBreak());
    dispatch(timerSlice.actions.stop());
  }
);
export const continueBreak = createAsyncThunk<void, void, { state: RootState }>(
  "continueBreak",
  (_, { dispatch, getState }) => {
    dispatch(currentTaskSlice.actions.continueBreak());
    dispatch(timerSlice.actions.start());
  }
);
export const nextOrContinueTask = createAsyncThunk<
  void,
  void,
  { state: RootState }
>("nextOrContinueTask", (_, { dispatch, getState }) => {
  dispatch(currentTaskSlice.actions.nextOrContinueTask());
});
export const completeTask = createAsyncThunk<void, void, { state: RootState }>(
  "completeTask",
  (_, { dispatch, getState }) => {
    // dispatch(task ) getState()
    dispatch(currentTaskSlice.actions.pauseBreak());
    // dispatch(timerSlice.actions.stop());
  }
);

const fakeTask = { id: -1, name: "", pomodoroCount: 0 };

export const currentTask = createSelector(
  currentTaskSlice.selectSlice,
  (state: RootState) => {
    const currentTask = currentTaskSlice.selectSlice(state);
    if (currentTask.taskId !== undefined) {
      return tasksSlice.selectors.getTask(state, currentTask.taskId);
    } else {
      return fakeTask;
    }
  },
  (currentTask, task) => {
    return {
      ...currentTask,
      ...task,
    };
  }
);

const mapStateToPrimaryName: {
  [key in CurrentTask["state"]]: string;
} = {
  inited: "Старт",
  workInited: "Старт",
  workTimer: "Пауза",
  workPause: "Cтарт",
  break: "Старт",
  breakTimer: "Пауза",
  breakStop: "Продолжить",
  workStop: "Продолжить",
};

const mapStateToPrimaryAction: {
  [key in CurrentTask["state"]]: (() => ActionCreator<void>) | undefined;
} = {
  inited: undefined,
  workInited: startTask,
  workTimer: pauseTask,
  workPause: continueTask,
  break: continueBreak,
  breakTimer: pauseBreak,
  breakStop: continueBreak,
  workStop: pauseTask,
};
const mapStateToSecondaryName: {
  [key in CurrentTask["state"]]: string;
} = {
  inited: "Стоп",
  workInited: "Стоп",
  workTimer: "Стоп",
  workPause: "Стоп",
  break: "Стоп",
  breakTimer: "Продолжить",
  breakStop: "Продолжить",
  workStop: "Сделано",
};

const mapStateToSecondaryAction: {
  [key in CurrentTask["state"]]: (() => ActionCreator<void>) | undefined;
} = {
  inited: undefined,
  workInited: undefined,
  workTimer: startBreak,
  workPause: startBreak,
  break: nextOrContinueTask,
  breakTimer: nextOrContinueTask,
  breakStop: nextOrContinueTask,
  workStop: completeTask,
};

export const taskActions = createSelector(
  (state: RootState) => state.currentTask.state,
  (currentState) => {
    return {
      primaryActionName: mapStateToPrimaryName[currentState],
      primaryAction: mapStateToPrimaryAction[currentState],
      secondaryActionName: mapStateToSecondaryName[currentState],
      secondaryAction: mapStateToSecondaryAction[currentState],
    };
  }
);

export function generateTasks(
  currentDate = new Date(),
  seed = 1
): SerialiazableTask[] {
  const totalMilisecondInDay = 1000 * 60 * 60 * 24;

  function random(max: number) {
    var x = Math.sin(seed++) * 10000;
    return Math.floor(max * (x - Math.floor(x)));
  }

  const result: SerialiazableTask[] = [];
  let id = -1;
  for (let i = 0; i < 7 * 3; i++) {
    const completedPomodoro = (random(4) + 1) * TASK_TIME;
    result.push({
      name: "Generated task " + i,
      id,
      startDateString: currentDate.toUTCString(),
      isCompleted: true,
      pomodoroCount: 3,
      completedPomodoro,
      workTime: Math.floor(
        completedPomodoro + completedPomodoro * (random(30) / 100)
      ),
      pauseCount: random(4),
      pauseTime: random(20 * 1000),
    });
    currentDate = new Date(currentDate.getTime() - totalMilisecondInDay);
    id = id - 1;
  }
  return result.reverse();
}

export function configureAppStore(preloadedState = {}) {
  const listenerMiddleware = createListenerMiddleware();
  listenerMiddleware.startListening({
    predicate: isAnyOf(addTask),
    effect: () => {
      console.log("Added___");
    },
  });
  const store = configureStore({
    reducer: rootReducer,
    preloadedState,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().prepend(listenerMiddleware.middleware),
  });
  if (process.env.NODE_ENV !== "production" && (module as any).hot) {
    (module as any).hot.accept("./reducers", () =>
      store.replaceReducer(rootReducer)
    );
  }
  return store;
}

type Store = ReturnType<typeof configureAppStore>;
// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = Store["dispatch"];
export const useAppDispatch = useDispatch.withTypes<AppDispatch>(); // Export a hook that can be reused to resolve types
