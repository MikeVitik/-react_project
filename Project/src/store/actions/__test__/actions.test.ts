import { CurrentTask } from "../../slicies/current-task";
import { Timer, timerSlice } from "../../slicies/timer-slice";
import { configureAppStore } from "../../store";
import {
  continueTask,
  createTask,
  pauseTask,
  startBreak,
  startTask,
} from "../task-actions";
import { updateTime } from "../update-time";

describe("taskActions", () => {
  describe("createTask", () => {
    it("check initial state", () => {
      const store = configureAppStore({});
      expect(store.getState().currentTask.state).toBe("inited");
      expect(store.getState().currentTask.primaryDisabled).toBe(true);
      expect(store.getState().currentTask.primaryAction).toBe("startTask");
      expect(store.getState().currentTask.secondaryDisabled).toBe(true);
      expect(store.getState().currentTask.secondaryAction).toBe("stopTask");
    });
    it("should add task, setCurrent task and timer", () => {
      const store = configureAppStore({});
      store.dispatch(createTask("New Task"));
      expect(store.getState().tasks[0].name).toBe("New Task");
      expect(store.getState().currentTask.taskId).toBe(
        store.getState().tasks[0].id
      );
      expect(store.getState().currentTask.state).toBe("workInited");
      expect(store.getState().timer.state).toBe("created");
      expect(store.getState().timer.totalTime).toBeGreaterThan(0);
    });
    it("should not create timer if task already started", () => {
      const store = configureAppStore({});
      store.dispatch(createTask("New Task"));
      store.dispatch(startTask());
      store.dispatch(updateTime(200));
      store.dispatch(createTask("New Task 2"));
      expect(store.getState().tasks[0].name).toBe("New Task");
      expect(store.getState().currentTask.taskId).toBe(
        store.getState().tasks[0].id
      );
      expect(store.getState().currentTask.state).toBe("workTimer");
      expect(store.getState().timer.state).toBe("running");
      expect(store.getState().timer.totalTime).toBeGreaterThan(200);
    });
  });

  describe("startTask", () => {
    it("should update cuttentTask and run timer", () => {
      const store = configureAppStore({});
      store.dispatch(createTask("New Task"));
      store.dispatch(startTask());

      expect(store.getState().currentTask.state).toBe("workTimer");
      expect(store.getState().currentTask.primaryAction).toBe("pauseTask");
      expect(store.getState().timer.state).toBe("running");
    });
  });
  describe("pauseTask", () => {
    beforeAll(() => {
      jest.useFakeTimers();
      jest.setSystemTime(new Date(2020, 3, 1, 10, 0, 0, 0));
    });

    afterAll(() => {
      jest.useRealTimers();
    });

    it("should add statistic item", () => {
      const store = configureAppStore({});
      store.dispatch(createTask("New Task"));
      store.dispatch(startTask());
      store.dispatch(timerSlice.actions.updateTime(3000));
      store.dispatch(pauseTask());
      expect(store.getState().currentTask.state).toBe("workPause");
      expect(store.getState().currentTask.hasPause).toBe(true);
      expect(store.getState().statisticInfo.length).toBe(1);
      expect(store.getState().statisticInfo[0]).toEqual({
        startDateString: "Wed, 01 Apr 2020 07:00:00 GMT",
        taskId: expect.any(Number),
        type: "work",
        workTime: 3000,
      });
    });

    it("should add pause statistic after continueTask", async () => {
      const store = configureAppStore({});
      store.dispatch(createTask("New Task"));
      store.dispatch(timerSlice.actions.updateTime(3000));
      store.dispatch(startTask());

      store.dispatch(timerSlice.actions.updateTime(3000));
      store.dispatch(pauseTask());
      store.dispatch(timerSlice.actions.updateTime(1500));

      store.dispatch(continueTask());
      expect(store.getState().currentTask.hasPause).toBe(true);
      expect(store.getState().statisticInfo.length).toBe(2);
      expect(store.getState().statisticInfo[1]).toEqual({
        startDateString: "Wed, 01 Apr 2020 07:00:00 GMT",
        taskId: expect.any(Number),
        type: "pause",
        pauseTime: 1500,
      });
    });
  });

  describe("break", () => {
    it("startBreak timer pause", () => {
      const store = configureAppStore({
        currentTask: { taskId: 0, state: "workPause" },
        timer: {
          time: 200,
          currentTime: 500,
          state: "pause",
          totalTime: 1000,
        } satisfies Timer,
        tasks: [{ id: 0, name: "New Task" }],
      });
      store.dispatch(startBreak());
      expect(store.getState().currentTask.state).toBe("break");
      expect(store.getState().timer.state).toEqual("created");
      expect(store.getState().statisticInfo[0]).toEqual({
        startDateString: expect.any(String),
        taskId: 0,
        type: "pause",
        pauseTime: 500,
      });
    });
    it("startBreak timer running", () => {
      const store = configureAppStore({
        currentTask: {
          taskId: 0,
          state: "workTimer",
          hasPause: true,
        } satisfies CurrentTask,
        timer: {
          time: 1000,
          currentTime: 500,
          state: "running",
          totalTime: 2000,
        },
        tasks: [{ id: 0, name: "New Task" }],
      });
      store.dispatch(startBreak());
      expect(store.getState().currentTask.state).toBe("break");
      expect(store.getState().timer.state).toEqual("created");
      expect(store.getState().statisticInfo[0]).toEqual({
        startDateString: expect.any(String),
        taskId: 0,
        type: "work",
        workTime: 500,
      });
    });
    it("startBreak from timer end", () => {
      const store = configureAppStore({
        currentTask: {
          taskId: 0,
          state: "workTimer",
        } satisfies CurrentTask,
        timer: { time: 1000, currentTime: 1000, state: "end", totalTime: 1000 },
        tasks: [{ id: 0, name: "New Task" }],
      });
      store.dispatch(startBreak());
      expect(store.getState().currentTask.state).toBe("break");
      expect(store.getState().timer.state).toEqual("created");
      expect(store.getState().statisticInfo[0]).toEqual({
        startDateString: expect.any(String),
        taskId: 0,
        completedPomodoro: 1,
        type: "work",
        workTime: 1000,
      });
    });
  });

  describe("completeTask", () => {
    it("should mark task completed", () => {});
    it("should increment completed_pomodoro", () => {});
    it("should move to initedTask state if has task", () => {});
    it("should move to inited state if has not task", () => {});
  });
});
