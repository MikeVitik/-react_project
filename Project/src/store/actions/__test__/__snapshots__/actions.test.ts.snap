// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taskActions break startBreak timer pause 1`] = `
Object {
  "startDateString": "Tue, 30 Apr 2024 16:52:27 GMT",
  "taskId": 0,
  "type": "worktime",
  "workTime": 500,
}
`;
